### PR TITLE
lxc/completion: `lxc storage volume copy` shell completion fixes and improvements to `completions.go`

### PR DESCRIPTION
**Pull request by kadinsayani**
Fixes https://github.com/canonical/lxd/issues/14264.

This PR also replaces the use of `Sprintf()` with string concatenation where possible, and adds go doc comments to all completion functions.

### CODE REVIEW

File: lxc/completion.go [hunk 1]
```diff
@@ -1174,13 +1174,14 @@ func (g *cmdGlobal) cmpStoragePoolWithVolume(toComplete string) ([]string, cobra
 
 	var results []string
 	for _, volume := range volumes {
-		results = append(results, fmt.Sprintf("%s/%s", pool, volume))
+		volName, _ := parseVolume("volume", volume)
+		results = append(results, fmt.Sprintf("%s/%s", pool, volName))
```
Comment (simondeziel) [inline lxc/completion.go:1178]: I know the code base is filled with this `fmt.Sprintf("%s/%s")` or other simple concatenation of strings but those are less efficient than a simple `+` operation when fancy conversion/formatting isn't required.

**Review by simondeziel — COMMENTED** [review]
I've yet to test it locally but it LGTM. I'll to get to it Tuesday.

File: lxc/completion.go (hunk 1 above)
Comment (kadinsayani) [reply lxc/completion.go:1178]: You're right! I'll update that :)

File: lxc/completion.go [hunk 2]
```diff
@@ -1174,13 +1174,14 @@ func (g *cmdGlobal) cmpStoragePoolWithVolume(toComplete string) ([]string, cobra
...
 	}
 
 	return results, cobra.ShellCompDirectiveNoFileComp
 }
 
-func (g *cmdGlobal) cmpStoragePools(toComplete string) ([]string, cobra.ShellCompDirective) {
+func (g *cmdGlobal) cmpStoragePools(toComplete string, noSpace bool) ([]string, cobra.ShellCompDirective) {
```
Comment (tomponline) [inline lxc/completion.go:1184]: Docblock missing explaining function and args

Comment (kadinsayani) [reply lxc/completion.go:1269]: Thanks for pointing this out - I've added doc comments to all the other completion functions as well (in a separate commit).

**Review by tomponline — APPROVED** [review]
Thanks!

**Pull request merged**